src/stk/_internal/topology_graphs/polymer/ncore/ncore.py

```diff
@@ -0,0 +1,551 @@
+import typing
+from collections import abc
+from dataclasses import dataclass
+
+import numpy as np
+
+from stk._internal.building_block import BuildingBlock
+from stk._internal.optimizers.null import NullOptimizer
+from stk._internal.optimizers.optimizer import Optimizer
+from stk._internal.reaction_factories.generic_reaction_factory import (
+    GenericReactionFactory,
+)
+from stk._internal.reaction_factories.reaction_factory import (
+    ReactionFactory,
+)
+from stk._internal.topology_graphs.edge import Edge
+from stk._internal.topology_graphs.topology_graph.topology_graph import (
+    TopologyGraph,
+)
+from stk._internal.topology_graphs.vertex import Vertex
+
+from ..vertices import CoreVertex, SubstituentVertex
+
+
+class NCore(TopologyGraph):
+    """
+    Represents a core reacting with N arms topology graph.
+
+    The core building block, with N functional groups, defines the
+    behaviour of this topology graph. The arm building blocks, which
+    must have only one functional group, are oriented in a circle around
+    the core based on N and the repeating units. The core is oriented
+    into the XY plane.
+
+    Examples:
+
+        *Construction*
+
+        The number of functional groups in the core building block
+        define the topology graph
+
+        .. testcode:: construction
+
+            import stk
+
+            core1 = stk.BuildingBlock(
+                smiles="C(Br)1C(Br)C1Br",
+                functional_groups=[stk.BromoFactory()],
```

Comment:
change examples to use `functional_groups=stk.BromoFactory()`
-- no need for the `[ ... ]` any more